src/shared/sol-util.c

```diff
@@ -86,6 +86,17 @@ sol_util_memdup(const void *data, size_t len)
     return ptr;
 }
 
+SOL_API long int
+sol_util_strtol(const char *nptr, char **endptr, ssize_t len, int base)
+{
+    char *tmpbuf;
+
+    if (len < 0)
+        len = (ssize_t)strlen(nptr);
+    tmpbuf = strndupa(nptr, len);
+    return strtol(tmpbuf, endptr, base);
```

Comment:
buggggggg...

endptr is pointing inside tmpbuf! You should calculate the offset and then make `endptr = nptr + offset`.